Services/Mail/classes/Preview/class.ilCourseMailTemplateTutorContextPreview.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright (c) 2017 ILIAS open source, Extended GPL, see docs/LICENSE
+ * CaT Concepts and Training GmbH
+ */
+include_once './Modules/Course/classes/class.ilCourseMailTemplateTutorContext.php';
+
+/**
+ * Preiview context for course mail template
+ *
+ * @author Stefan Hecken 	<concepts-and-training.de>
+ */
+class ilCourseMailTemplateTutorContextPreview extends ilCourseMailTemplateTutorContext {
+	const ID = 'crs_context_tutor_manual_preview';
+
+	const DEFAULT_COURSE_TITLE = "preview_crs_title";
+	const DEFAULT_COURSE_LINK = "preview_crs_link";
+	const DEFAULT_COURSE_STATUS = "preview_crs_status";
+	const DEFAULT_COURSE_MARK = "preview_crs_mark";
+	const DEFAULT_COURSE_TIME_SPENT = "3671";
+
+	public function __construct() {
+		global $DIC;
+		$this->g_lng = $DIC->language();
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getId()
+	{
+		return self::ID;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function resolveSpecificPlaceholder($placeholder_id, array $context_parameters, ilObjUser $recipient = null, $html_markup = false)
+	{
+		if(!in_array($placeholder_id, array('crs_title', 'crs_link')))
+		{
+			return "";
+		}
+
+		$this->g_lng->loadLanguageModule('sess');
+		$ret = null;
+		switch($placeholder_id)
```

Comment:
Wo soll dieses Preview erscheinen? Im Kurs? Kann man sich dann nicht direkt darauf beziehen?

Reply:
P.S.: Warum übergibt man die Platzhalter nicht über die kontext-parameter?

Reply:
zu 1) Die Preview erscheint bei der Anlage von der Vorlage.
zu 2) Welche Platzhalter willst du übergeben? Hier werden die Platzhalter im Text durch passende Werte ersetzt. Der Platzhalter der ersetzt werden soll, versteckt sich in $placeholder_id. Ich muss ja schon wissen was bei welchem Platzhalter passieren soll. Wenn die Möglichen Platzhalter über $context_parameter kommen, müsste ich ja noch mehr "switchen"